languages/bash.scm

```diff
@@ -1,2 +1,114 @@
 ; Configuration
 (#language! bash)
+
+; Don't modify string literals
+(string) @leaf
+
+; Allow blank line before
+; TODO Add directives here...
+[
+  (comment)
+  (if_statement)
+] @allow_blank_line_before
+
+; Prepend spaces
+; (Not "surround", to avoid extraneous spaces between delimiters)
+; TODO Add keywords, etc., here...
+[
+  "if"
+  "then"
+  "elif"
+  "else"
+  "fi"
+  (string)
+  (test_command)
+] @prepend_space
+
+;; Commands
+
+; One command per line, modulo many exceptions:
+; * Strings of && and ||
+; * Pipelines
+; * Subshells
+; * As arguments to if, while, for, etc.
+; * Others?...
+;
+; FIXME Can this be generalised, or does *every* context need to be
+; individually enumerated?...
+(program (command) @prepend_spaced_softline @append_hardline)
+(if_statement _ "then"(command) @prepend_spaced_softline)
+(elif_clause _ "then" (command) @prepend_spaced_softline)
+(else_clause (command) @prepend_spaced_softline)
+
+; Space between command line arguments
+(command
+  argument: _ @append_space @prepend_space
+)
+
+;; Operators
+
+; Ensure the negation operator is surrounded by spaces
+(negated_command
+  "!" @prepend_space @append_space
+)
+
+;; Conditionals
+
+; Start conditional on a new line
+[
+  (if_statement)
+  (elif_clause)
+  (else_clause)
+] @prepend_spaced_softline
+
+; New line after "then" and start indent block
+[
+  (if_statement)
+  (elif_clause)
+] "then" @append_hardline @append_indent_start
+
+; New line after "else" and start indent block
+(else_clause "else" @append_hardline @append_indent_start)
+
+; Keep the "if" and the "then" on the same line,
+; inserting a delimiter when necessary
+(if_statement
+  (_) @prepend_space @append_delimiter
+  ";"* @do_nothing
+  "then"
+
+  (#delimiter! ";")
+)
+
+; FIXME Can this be generalised from the above?
```

Comment:
Maybe if you say "anything that can have `then` in it":

```
(_
  (_) @prepend_space @append_delimiter
  ";"* @do_nothing
  "then"

  (#delimiter! ";")
)
```

Reply:
Nice :+1: